entitymanager.js

```diff
@@ -0,0 +1,44 @@
+function EntityManager() {
+  var entities = {};
+  var nextId = 0;
+
+  this.add = function(entity) {
+    entity.registerId(nextId);
+    entities[nextId++] = entity;
+  }
+
+  this.update = function() {
+    for (var index in entities) {
+      if (entities[index].update()) {
+        // Deletes the index property from the entities object.
+        delete entities[index];
+      }
+    }
+  }
+
+  this.render = function() {
+    for (var index in entities) {
+      entities[index].render();
+    }
+  }
+
+  this.checkCollisions = function() {
+    var x = 1;
+    for (var i in entities) {
```

Comment:
No problem, if you can think of a better way to loop through this without getting duplicate pairs and not changing entities to an array do tell me :)

P.S. doing the for loop with the indices would work if you used the nextId variable but you would get alot of undefined properties and that gets worse and worse as we play the game longer.